src/development/troubleshoot.md

```diff
@@ -140,15 +140,24 @@ WARNING: [pool web] server reached max_children setting (2), consider raising it
 
 That indicates that the server is receiving more concurrent requests than it has PHP processes allocated, which means some requests will have to wait until another finishes.  In this example there are 2 PHP processes that can run concurrently.
 
-Platform.sh sets the number of workers based on the available memory of your container and the estimated average memory size of each process.  There are two ways to increase the number of workers:
+The following command will identify the 20 slowest requests in the last hour, which can provide an indication of what code paths to investigate.
 
-* Adjust the [worker sizing hints](/languages/php/fpm.md) for your project.
-* Upgrade your subscription on Platform.sh to get more computing resources. To do so, log into your [account](https://accounts.platform.sh) and edit the project.
+```bash
```

Comment:
I don't agree with collapsing these two.  Talking about "server reached max_children" without pointing people at the worker sizing seems pointless.  It's one of the more straightforward things to do, and if a process has a lot of non-static callbacks on a given page then every single request is going to end up blocking on max_children on every request.  Jumping from that straight to finding slow requests seems like a non-sequitor.

Reply:
See a883dc6daf548dd08494975ee495765714628ca4, that's the intention for adjusting number of workers in moat case.

When the application is always hitting the limit, it should upgrade subscription / make better caching.

When the workload is CPU bound, adding more FPM processes without adding computing resources will just queue requests in FPM (instead of Nginx), waiting for CPU to be available.

If the workload is I/O bound, that's another story. Adding more workers may help non I/O bound request getting a chance for processing but all FPMs will eventually be filled if the I/O is slow.